dmq-node/app/Main.hs

```diff
@@ -69,48 +79,68 @@ runDMQ commandLineConfig = do
 
     stdGen <- newStdGen
     let (psRng, policyRng) = split stdGen
+        diffusionTracers = dmqDiffusionTracers dmqConfig tracer
+
+    Diffusion.withIOManager \iocp -> do
+      let localSnocket' = localSnocket iocp
+
+      withNodeKernel @StandardCrypto psRng $ \nodeKernel -> do
+        dmqDiffusionConfiguration <- mkDiffusionConfiguration dmqConfig nt
+
+        let stakePoolMonitor = connectToCardanoNode tracer localSnocket' snocketPath nodeKernel
```

Comment:
The `stakePoolMonitor` thread should be started by `withNodeKernel` since it writes to a `TVar` which is part of `NodeKernel`.